comp/otelcol/converter/def/component.go

```diff
@@ -12,12 +12,7 @@ import (
 
 // team: opentelemetry
 
-// Component implements the confmap.Converter interface and
-// provides extra functions to expose the provided and enhanced configs.
+// Component implements the confmap.Converter interface.
 type Component interface {
 	confmap.Converter
```

Comment:
Could we try removing the direct reference to the third part dependency as part of the component interface? 

ideally, we should not reference third part dependencies directly into the component interface, that way if in the future someone wants to replace the converter with a separate implementation that do not depend on `"go.opentelemetry.io/collector/confmap"` they would be able to.

Reply:
I could switch it to:
```
type Component interface {
	Convert(ctx context.Context, conf *confmap.Conf) error
}
```
But this also relies on `go.opentelemetry.io/collector/confmap`.

---

Alternatively, I could define the component as so:
```
// Component implements the confmap.Converter interface.
type Component interface{}
```

and impl as so:
```
type DDConverter struct {}

var (
	_ confmap.Converter = (*DDConverter)(nil)
)

// NewConverter currently only supports a single URI in the uris slice, and this URI needs to be a file path.
func NewConverter() (converter.Component, error) {
	return &DDConverter{}, nil
}

// Convert autoconfigures conf and stores both the provided and enhanced conf.
func (c *DDConverter) Convert(_ context.Context, conf *confmap.Conf) error {
	enhanceConfig(conf)
	return nil
}
```

and then do some type assertions:
```
			conv := converter.(DDConverter)
			conv.Convert(context.Background(), conf)
```
but this seems a bit hacky.

Is there another approach that you had in mind ?

---

As a note, this component is made to fit in within the collectors APIs, which is why it implements upstream APIs -- but I understand the point of not wanting third part dependencies.

Reply:
We want to avoid type casting here as the goal is to have a clear interface for user.

The question here is: would it make sense at some point to have a converter that work with other types than `go.opentelemetry.io` ? If so we should abstract the types take into the public interface, if not we can directly use OTEL types.

As an example, if makes sense for a `docker` component to user docker types. But a `container` component would abstract types so we could have different implementation behind.

Reply:
@hush-hush 

> The question here is: would it make sense at some point to have a converter that work with other types than go.opentelemetry.io ? If so we should abstract the types taken by the public interface, if not we can directly use OTEL types.

I don't see a case where we would do this type of conversion on another type than `go.opentelemetry.io`, as this component is made to be passed within [ResolverSettings](https://pkg.go.dev/go.opentelemetry.io/collector/confmap#ResolverSettings) (`go.opentelemetry.io`), and all the logic it contains uses `go.opentelemetry.io/collector/confmap` (e.g. [see example file](https://github.com/DataDog/datadog-agent/blob/main/comp/otelcol/converter/impl/extensions.go)). It reads and converts otel configs in that format.

Reply:
I think keeping `confmap.Converter` is probably better than using type assertions in this case.

Reply:
Agree with the assessment. Ideally all our component interface would be free of third party dependencies, but that would increase the complexity of the code base by interface proliferation. Still, there will be cases that is worth extracting the third party dependency in an interface, but for this particular case as @mackjmr alluded there is little chances that this component is used outside of the OTel ecosystem.

Reply:
I agree with @ogaca-dd here, this component is intimately tied to the opentelemetry collector and alternate implementations would also be geared for integration in the opentelemetry collector, thus, this reference to a third-party dep is not _that_ problematic given the scope.